functions/Remove-DBAQueryPlan.ps1

```diff
@@ -0,0 +1,96 @@
+﻿function Remove-DbaQueryPlan {
+<#
+    .SYNOPSIS
+        Removes adhoc and prepared plan caches is single use plans are over defined 
+        threshold
+
+    .DESCRIPTION
+        "Checks ahoc and prepared plan cache for each database, if over 100 MBs 
+        removes from the cache."
+
+        This command automates that process.
+
+        References: https://www.sqlskills.com/blogs/kimberly/plan-cache-adhoc-workloads-and-clearing-the-single-use-plan-cache-bloat/
+
+        Note: This command removes the plans from all SQL instances on the destionation 
+        server but the process column is specific to -SqlInstance passed.
+
+    .PARAMETER SqlInstance
+        Allows you to specify a comma separated list of servers to remove the plans.
+
+    .PARAMETER SqlCredential
+        Allows you to login to the SQL instance using alternative credentials.
+
+    .PARAMETER Threshold
+        Memoery used threshold.
+
+    .PARAMETER WhatIf
+        If this switch is enabled, no actions are performed but informational messages will be displayed that explain what would happen if the command were to run.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Memory
+        Author: Tracy Boggiano, databasesuperhero.com
+
+        dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+        Copyright (C) 2016 Chrissy LeMaire
+        License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+    .LINK
+        https://dbatools.io/Remove-DbaQueryPlan
+
+    .EXAMPLE
+        Remove-DbaQueryPlan -SqlInstance sql2017 -Threshold 200
+
+        Logs into the SQL Server instance "sql2017" and removes plan caches if over 200 MBs.
+
+    .EXAMPLE
+        Remove-DbaQueryPlan -SqlInstance sql2017 -SqlCredential (Get-Credential sqladmin) 
+
+        Logs into the SQL instance using the SQL Login 'sqladmin' and then Windows instance as 'ad\sqldba'
+        and removes if Threshold over 100 MBs.
+#>
+    [CmdletBinding()]
```

Comment:
Since this is a remove command we need to add support for WhatIf processing...You can see an example of this here: https://github.com/sqlcollaborative/dbatools-templates/blob/master/Remove-Set-Write-DbaWhatever.ps1